src/components/common/time-range-slider.js

```diff
@@ -119,9 +118,9 @@ export default class TimeRangeSlider extends Component {
   };
 
   render() {
-    const {domain, value, isEnlarged} = this.props;
+    const {domain, value, isEnlarged, onChange} = this.props;
     const {isAnimating} = this.state;
-
+    this._sliderThrottle = throttle(onChange, 20);
```

Comment:
This creates a new `_sliderThrottle` even if `onChange` is the same. It may cause many unnecessary re-renders.
How about saving the `onChange` in the instance and creating a new `_sliderThrottle` only as needed (memoization)?

Reply:
since onChange is passed in as a lambda. so it's always a different lambda when parent widget renders. Is it really necessary or even possible to add that optimization?

Reply:
That's an excellent point. In that case, how about using a lambda for calling `this.props.onChange` and leaving this constructed once in the constructor?